src/importexport/guitarpro/internal/guitarbendimport/benddatacollector.cpp

```diff
@@ -0,0 +1,286 @@
+
+    auto pitchDiff = [](int prevPitch, int currentPitch) {
+        if (prevPitch == currentPitch) {
+            return PitchDiff::SAME;
+        }
+
+        return (prevPitch < currentPitch) ? PitchDiff::UP : PitchDiff::DOWN;
+    };
+
+    if (adaptedPitchValues.front().pitch != 0 && !note->tieBack()) {
+        BendSegment seg;
+        seg.startTime = seg.endTime = 0;
+        seg.pitchDiff = adaptedPitchValues.front().pitch;
+
+        bendSegments.push_back(seg);
+    }
+
+    for (size_t i = 0; i < adaptedPitchValues.size() - 1; i++) {
+        currentPitchDiff = pitchDiff(adaptedPitchValues[i].pitch, adaptedPitchValues[i + 1].pitch);
+        if (currentPitchDiff == previousPitchDiff) {
+            if (!bendSegments.empty()) {
+                BendSegment& lastSeg = bendSegments.back();
+                lastSeg.endTime = adaptedPitchValues[i + 1].time;
+                lastSeg.pitchDiff = adaptedPitchValues[i + 1].pitch - adaptedPitchValues[i].pitch;
+            }
+
+            continue;
+        }
+
+        if (currentPitchDiff == PitchDiff::SAME) {
+            BendSegment seg;
+            seg.startTime = adaptedPitchValues[i].time;
+            seg.endTime = adaptedPitchValues[i + 1].time;
+            seg.pitchDiff = adaptedPitchValues[i + 1].pitch - adaptedPitchValues[i].pitch;
+            bendSegments.push_back(seg);
+        } else {
+            if (previousPitchDiff != PitchDiff::SAME || bendSegments.empty()) {
+                BendSegment seg;
+                seg.startTime = adaptedPitchValues[i].time;
+                seg.endTime = adaptedPitchValues[i + 1].time;
+                bendSegments.push_back(seg);
+            } else {
+                BendSegment& lastSeg = bendSegments.back();
+                lastSeg.middleTime = adaptedPitchValues[i].time;
+                lastSeg.endTime = adaptedPitchValues[i + 1].time;
+            }
+
+            bendSegments.back().pitchDiff = adaptedPitchValues[i + 1].pitch - adaptedPitchValues[i].pitch;
+        }
+
+        previousPitchDiff = currentPitchDiff;
+    }
+
+    ImportedBendInfo info;
+    info.segments = std::move(bendSegments);
+    info.note = note;
+
+    for (const auto& bs : info.segments) {
+        if (bs.pitchDiff != 0 && bs.startTime != bs.endTime) {
+            info.pitchChangesAmount++;
+        }
+    }
+
+    return info;
+}
+
+static bool isSlightBend(const BendDataCollector::ImportedBendInfo& importedInfo)
+{
+    if (importedInfo.pitchChangesAmount != 1 || importedInfo.note->tieFor() || importedInfo.segments.size() != 2) {
+        return false;
+    }
+
+    for (const auto& seg : importedInfo.segments) {
+        if (seg.pitchDiff == 25) {
+            return true;
+        }
+    }
+
+    return false;
+}
+
+void fillSlightBendDataContext(BendDataContext& bendDataCtx, const BendDataCollector::ImportedBendInfo& importedInfo)
+{
+    Note* note = importedInfo.note;
+    Chord* chord = note->chord();
```

Comment:
const Note*, const Chord*

Reply:
thanks @mikekirin, your const attention to const helped me to find non const methods in dom:
<img width="439" alt="Screenshot 2024-12-18 at 18 49 21" src="https://github.com/user-attachments/assets/a0b70663-ff9b-4d49-89b0-37dc42b09219" />